crates/query-engine/translation/src/translation/error.rs

```diff
@@ -201,3 +201,30 @@ impl std::fmt::Display for Error {
         }
     }
 }
+
+/// A type for translation warnings.
+#[derive(Debug, Clone)]
+pub enum Warning {
```

Comment:
I didn't realise you meant creating an enum for `Warning`, but this is very good and means moving to a more formal warnings API will be easier.

Reply:
I didn't plan on it at first, but I had to use the same warning in multiple places, and it just makes it easier to have an enum in that case.